src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -1706,6 +1716,42 @@ PeerImp::checkValidation (Job&, STValidation::pointer val,
     }
 }
 
+// Returns the set of peers that can help us get
+// the TX tree with the specified root hash.
+//
+static
+std::vector<std::shared_ptr<PeerImp>>
+getPeersWithTree (OverlayImpl& ov,
+    uint256 const& rootHash, PeerImp const* skip)
+{
+    std::vector<std::shared_ptr<PeerImp>> v;
```

Comment:
Hmm...okay. But note that `ov.size()` takes a lock and the number might change. Also, this code will likely be replaced by David's changes. Still, why not throw it in there?